papers/bradley_dice/55_bradley_dice.rst

```diff
@@ -0,0 +1,564 @@
+
+The synced collections subpackage of **signac** represents the culmination of our efforts to expose this functionality, providing a generic framework within which interfaces corresponding to any of Python's built-in types can be easily constructed with arbitrary underlying synchronization protocols.
+For instance, with synced collections it becomes easy to define a new list-like type that automatically saves all its data in a plain-text CSV format.
+However, the flexibility of this new framework extends far beyond that, defining a generic protocol that can be used to provide a dictionary, list, or set-like API to any arbitrary underlying data structure, including other in-memory objects that do not present a similarly Pythonic API.
+
+Summary of Features
+~~~~~~~~~~~~~~~~~~~
+
+We designed synced collections to be flexible, easily extensible, and independent of **signac**'s data model.
+The central element is the ``SyncedCollection`` class, which defines a new abstract class extending the ``collections.abc.Collection`` from the Python standard library.
+A ``SyncedCollection`` is a ``Collection`` that adds two additional groups of abstract methods that must be implemented by its subclasses.
+One group includes methods for synchronizing with an underlying resource, while the other contains methods for synchronizing with a standard collection of the underlying base type.
+For instance, a ``JSONDict`` would implement the first set of methods to define how to save a dictionary to a JSON file and reload it, while it would implement the second set of methods to define how to convert between a ``JSONDict`` instance and a standard Python dictionary.
+
+Critically, these two sets of functions are orthogonal.
+Therefore, it is possible to implement different backend types and different data structures independently, then combine them into concrete classes using multiple inheritance.
+This solution is analogous to the way that language server protocols separate support for programming languages from support for editors, turning an :math:`M \times N` problem into a simpler :math:`M+N` problem.
+In practice, our synced collections framework comes bundled with a set of backend classes, such as the ``JSONCollection``, and a set of data structure classes, such as the ``SyncedDict``.
+Each of these inherits from ``SyncedCollection`` and implements a subset of its methods, but remains abstract until combined (via multiple inheritance) with a class implementing the remaining methods.
+This design pattern makes defining the functional classes at the bottom of the hierarchy trivial.
+For example, the ``JSONDict`` is implemented by inheriting from ``JSONCollection`` and ``SyncedDict``, but requires no additional code to function.
+Similarly, the ``JSONList`` class inherits from ``JSONCollection`` and ``SyncedList``.
```

Comment:
This portion also has a heavy focus on implementation and seems more relevant to developers than casual users, whereas the rest of the paper seems to target casual users.